### PR TITLE
Implement Droplet GET /v3/droplets/:guid endpoint

### PR DESCRIPTION
**Pull request by akrishna90**
## Is there a related GitHub Issue?
#54 

## What is this change about?
Implements droplet GET endpoint.

## Does this PR introduce a breaking change?
No.

## Acceptance Steps
Follow acceptance steps on #54

### CODE REVIEW

File: apis/droplet_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		})
+		When("building the k8s client errors", func() {
+			BeforeEach(func() {
+				clientBuilder.Returns(nil, errors.New("boom"))
+				router.ServeHTTP(rr, req)
+			})
+
+			itRespondsWithUnknownError(getRR)
+		})
+		When("build staging fails", func() {
```
Comment (matt-royal) [inline apis/droplet_handler_test.go:164]: Since we're abstracting how the Droplet is represented with the DropletRepo, I'd rephrase this to "the droplet doesn't exist". That's more direct.

Comment (matt-royal) [reply apis/droplet_handler_test.go:164]: Actually, now that I've read further I think you should just delete this case and the one below about "staging is not complete". The setup for all 3 is identical and it doesn't matter *WHY* you're in that scenario, just that you are.

File: apis/droplet_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			BeforeEach(func() {
+				dropletRepo.FetchDropletReturns(repositories.DropletRecord{}, repositories.NotFoundError{})
+				router.ServeHTTP(rr, req)
+			})
+			itRespondsWithNotFound("Droplet not found", getRR)
+		})
+		When("build staging is not complete", func() {
```
Comment (matt-royal) [inline apis/droplet_handler_test.go:171]: Delete this as per the comment above.

File: apis/droplet_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+			BeforeEach(func() {
+				dropletRepo.FetchDropletReturns(repositories.DropletRecord{}, repositories.NotFoundError{})
+				router.ServeHTTP(rr, req)
+			})
+			itRespondsWithNotFound("Droplet not found", getRR)
+		})
+		When("the droplet cannot be found", func() {
```
Comment (matt-royal) [inline apis/droplet_handler_test.go:178]: Keep this one, delete the two cases above.

File: presenter/droplet.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		State:     dropletRecord.State,
+		Lifecycle: Lifecycle{
+			Type: dropletRecord.Lifecycle.Type,
+			Data: LifecycleData{
+				Buildpacks: dropletRecord.Lifecycle.Data.Buildpacks,
+				Stack:      dropletRecord.Lifecycle.Data.Stack,
+			},
+		},
+		ExecutionMetadata: "",
+		Buildpacks:        make([]BuildpackData, 0),
```
Comment (matt-royal) [inline presenter/droplet.go:53]: FYI, this could also be `[]BuildpackData{}`, which I think reads a little better.

File: presenter/droplet.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+				HREF: prefixedLinkURL(baseURL, fmt.Sprintf("v3/packages/%s", dropletRecord.PackageGUID)),
+			},
+			"app": &Link{
+				HREF: prefixedLinkURL(baseURL, fmt.Sprintf("v3/apps/%s", dropletRecord.AppGUID)),
+			},
+			"assign_current_droplet": &Link{
+				HREF:   prefixedLinkURL(baseURL, fmt.Sprintf("v3/apps/%s/relationships/current_droplet", dropletRecord.AppGUID)),
+				Method: "PATCH",
+			},
+			"download": nil,
```
Comment (matt-royal) [inline presenter/droplet.go:81]: I thought we decided to set this, even though the endpoint isn't implemented. If we don't set it now, we'll have to come back in the future and add it.

File: repositories/droplet_repository.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	stagingStatus := getConditionValue(&cfBuild.Status.Conditions, StagingConditionType)
+	succeededStatus := getConditionValue(&cfBuild.Status.Conditions, SucceededConditionType)
+	if stagingStatus == metav1.ConditionFalse &&
+		succeededStatus == metav1.ConditionTrue {
+		return r.cfBuildToDropletRecord(cfBuild), nil
+	}
+	return DropletRecord{}, NotFoundError{}
+}
+
+func (r *DropletRepo) cfBuildToDropletRecord(cfBuild workloadsv1alpha1.CFBuild) DropletRecord {
```
Comment (matt-royal) [inline repositories/droplet_repository.go:61]: This doesn't use any of the information on the DropletRepo, so it should be a message without a receiver.

File: repositories/droplet_repository_test.go [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("DropletRepository", func() {
+	Describe("GetDroplet", func() {
```
Comment (matt-royal) [inline repositories/droplet_repository_test.go:21]: The method is called `FetchDroplet`, which doesn't match. We've used Fetch elsewhere, so we should probably change this line and not rename the method.

File: apis/droplet_handler_test.go [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/rest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ = Describe("DropletHandler", func() {
+	const (
+		testDropletHandlerLoggerName = "TestDropletHandler"
+	)
+	Describe("the GET /v3/droplet/:guid endpoint", func() {
```
Comment (matt-royal) [inline apis/droplet_handler_test.go:23]: It's a minor thing, but I find it hard to read test files where there's no newlines between sibling `It`, `When`, etc. statements, like in this file. You don't need to change it for me, but I'll definitely add them if I'm ever working in this file. Maybe I should start a team discussion about this, since I know you're not the only one who leaves out the newlines and I may be in the minority here.

File: apis/droplet_handler_test.go (hunk 3 above)
Comment (akrishna90) [reply apis/droplet_handler_test.go:173]: Addressed on commit [8a5dfced](https://github.com/cloudfoundry/cf-k8s-api/pull/131/commits/8a5dfced2f8a16a0d25310f384e2cb265d2a4a7d)

File: presenter/droplet.go (hunk 4 above)
Comment (akrishna90) [reply presenter/droplet.go]: Addressed on commit [8a5dfced](https://github.com/cloudfoundry/cf-k8s-api/pull/131/commits/8a5dfced2f8a16a0d25310f384e2cb265d2a4a7d)

File: repositories/droplet_repository.go (hunk 6 above)
Comment (akrishna90) [reply repositories/droplet_repository.go]: Addressed on commit [8a5dfced](https://github.com/cloudfoundry/cf-k8s-api/pull/131/commits/8a5dfced2f8a16a0d25310f384e2cb265d2a4a7d)

File: repositories/droplet_repository_test.go (hunk 7 above)
Comment (akrishna90) [reply repositories/droplet_repository_test.go]: Addressed on commit [8a5dfced](https://github.com/cloudfoundry/cf-k8s-api/pull/131/commits/8a5dfced2f8a16a0d25310f384e2cb265d2a4a7d)

File: apis/droplet_handler_test.go (hunk 8 above)
Comment (akrishna90) [reply apis/droplet_handler_test.go:23]: Have addressed it. I don't have a strong inclination. I would like to hear from others 👍🏽

File: presenter/droplet.go (hunk 5 above)
Comment (akrishna90) [reply presenter/droplet.go:81]: I will address the change if required as a new PR. I'll merge as is.
